python/cudnn/gemm/frost/kernel_templates/sm107_block_scale_matmul_2ctamma.py

```
@@ -0,0 +1,1520 @@
+            sfb_tmem_bases = [(base_row_id << 16) | (base_col_id_root + sfb_col_bases[j]) for j in range(num_b_operands)]
+            s2t_shape, s2t_multicast = nvvm.S2TCopyMode.S2T_32x128b_WARPX4
+            sfa_scale_ptrs = [nvvm.make_tmem_ptr(b, cutlass.Float32) for b in sfa_tmem_bases]
+            sfb_scale_ptrs = [nvvm.make_tmem_ptr(b, cutlass.Float32) for b in sfb_tmem_bases]
+            # utccp destination per (MN-block, atom within the scale word). A
+            # word is atom-MAJOR across the blocks — atom ``a`` of block ``m``
+            # sits at ``(a*num_blocks + m)*registers_per_atom``, which is what
+            # the MMA's scale operand expects once a word spans more than one
+            # atom. At word_atoms == 1 this is sm100's ``m * registers_per_block``.
+            sfa_dst_ptrs = [
+                [
+                    [nvvm.make_tmem_ptr(sfa_tmem_bases[i] + m * registers_per_block + a * registers_per_atom, cutlass.Float32) for a in range(word_atoms)]
```

Ruff (0.16.1)

[error] Undefined name `registers_per_block`

(F821)